assets/aml-benchmark/components/src/aml_benchmark/utils/io.py

```diff
@@ -115,14 +115,16 @@ def resolve_io_path(dataset: str) -> List[str]:
 
 def read_jsonl_files(file_paths: List[str]) -> List[Dict[str, Any]]:
     """
-    Read `.jsonl` files and return a list of dictionaries.
+    Read `.jsonl` files if extendion is jsonl else calls `read_files` function for files \
+    that do not endwith jsonl extension but the file content is of type json.
 
-    Ignores files that do not have `.jsonl` extension. Raises exception if no `.jsonl` files
-    found or if any `.jsonl` file contains invalid JSON.
+    Return a list of dictionaries.
 
     :param file_paths: List of paths to .jsonl files.
     :return: List of dictionaries.
     """
+    if len(file_paths) == 1 and not file_paths[0].endswith(".jsonl"):
+        return read_files(file_paths)
     data_dicts = []
     for file_path in file_paths:
```

Comment:
nit: Can we use read_files function here? Example-
```
data_dicts = read_files([file_path for file_path in file_paths if file_path.endswith(".jsonl")])
```

Reply:
@arun-rajora Please elaborate more.

Reply:
read_files function is doing similar to what we have starting from line 129. We can simplify all of this using the approach mentioned above. Thanks!

Reply:
Made the changes as discussed on the call and removed the read_files function.